clang/include/clang/Sema/Sema.h

```diff
@@ -710,9 +710,11 @@ class Sema final {
     return result;
   }
 
+  // Saves the current floating-point pragma stack and clear it in this Sema.
   class FpPragmaStackSaveRAII {
   public:
-    FpPragmaStackSaveRAII(Sema &S) : S(S), SavedStack(S.FpPragmaStack) {}
+    FpPragmaStackSaveRAII(Sema &S)
+        : S(S), SavedStack(std::move(S.FpPragmaStack)) {}
```

Comment:
Does a move of `PragmaStack` guarantee to leave an empty stack?  I guess it's ultimately just a `SmallVector`, and it looks like `SmallVector` does currently make that guarantee.  That seems worth a comment or an assertion.

Reply:
Added explicit stack reset.